src/docs/database-migrations.mdx

```diff
@@ -376,3 +376,60 @@ Renaming columns is dangerous and will result in downtime. The reason this occur
   - Stop writing to the old column and remove references from the code.
   - Drop the old column from the database.
   - Generally, this is not worth doing and a lot of risk/effort compared to the reward.
+
+## Migration Deployment
+
+<Alert title="Employees Only" level="warning">
+  The following section covers migration deployment in Sentry's
+  saas products.
+</Alert>
+
+We support two kinds of migrations in our saas deployments:
+
+- Deploy time migrations
+- Post-deploy migrations
+
+### Deployment Migrations
+
+Deployment migrations are run in each region and tenant before code is deployed. Deployment migrations are expected to finish quickly and all statements must complete within 5 seconds. If a migration could take longer because a large number of rows is being operated on, it should be deployed as a post-deploy migration instead. Deployment migrations are ideal for:
+
+* Adding new tables and columns.
+* Adding indexes to most tables.
+* Removing columns and tables - as long as you follow the processes outlined
+  above.
+
+### Post-deploy migrations (is_dangerous) 
+
+Post-deploy migrations are run manually by engineers **after** a migration has been deployed **to all** regions. During deployment, post-deploy migrations are marked as complete (faked) with django's fake migration behavior. When a post-deploy migration is run, it is run against all regions and tenants. Post-deploy migrations are triggered manually by engineers.
+
+Post-deploy migrations are ideal for:
```

Comment:
Could we add a warning here to never delete/add columns, or do any other ddl changes other than adding indexes and constraints via post-deploy migration? This is an easy way people can accidentally cause an outage.

I actually wonder if we should build this into the tool as well.